core/src/main/scala/org/http4s/MediaType.scala

```diff
@@ -71,27 +78,77 @@ sealed class MediaRange private[http4s] (
   }
 }
 
-object MediaRange extends Registry {
-  type Key = String
-  type Value = MediaRange
-
-  implicit def fromKey(k: String): MediaRange = {
-    val parts = k.split('/')
-    if (parts.length < 2) new MediaRange(parts(0))
-    else if (parts(1) != "*") throw new IllegalArgumentException(k + " is not a valid media-type")
-    else new MediaRange(parts(0))
+object MediaRange {
+  val `*/*` = new MediaRange("*")
+  val `application/*` = new MediaRange("application")
+  val `audio/*` = new MediaRange("audio")
+  val `image/*` = new MediaRange("image")
+  val `message/*` = new MediaRange("message")
+  val `multipart/*` = new MediaRange("multipart")
+  val `text/*` = new MediaRange("text")
+  val `video/*` = new MediaRange("video")
+
+  val standard: Map[String, MediaRange] =
```

Comment:
I would have thought this includes media types without the second wild card, but I guess it didn't before?

Reply:
This is basically replaces the registry with the same set of registered media types as before. Should be fairly mechanical

Reply:
Hmm.  These aren't what I expected them to be.  I thought it was keyed by `mainType/subType` instead of just `mainType`.  I wonder why that is.  But you didn't change it.

Reply:
Well the `MediaRange` case class only has a `mainType` unlike `MediaType`
https://github.com/http4s/http4s/blob/master/core/src/main/scala/org/http4s/MediaType.scala#L97

What surprised me is that the extensions are ignored to set the key

Reply:
A MediaType is a MediaRange, so I'd expect to be able to look up MediaTypes via the MediaRange companion.

I'm not sure what I think about extensions.

Reply:
Any ideas to move this forward?